File: Theory of Algs/C#/PA2CS/C#/HuffmanDecoder.cs
using System;
using System.Collections;
namespace _PA2
{
    public class HuffmanDecoder
    {
        public static string decode(string encodedMsg, Hashtable encodingToCharMapping)
        {// complete this method
            string decodedMsg = "";
            string encode = "";
            int n = encodedMsg.Length;

            for (int i = 0; i < n; i++)
            {
                encode += encodedMsg[i];
                if (encodingToCharMapping.ContainsKey(encode))
                {
                    char c = (char) encodingToCharMapping[encode];
                    decodedMsg += c;
                    encode = "";
                }
            }

            return decodedMsg; // build into string value and return
        }

    }
}

File: Theory of Algs/Java/PA2 - edited/TheoryofAlgsPA2/src/HuffmanDecoder.java


import java.util.Hashtable;

public class HuffmanDecoder {

	public static String decode(String encodedMsg, Hashtable<String, Character> encodingToCharMapping) {
		String code = new String();
		String decoded = new String();
		int n = encodedMsg.length();
		for (int i = 0 ; i <= n-1 ; i++)
		{
			code += encodedMsg.charAt(i);
			if (encodingToCharMapping.containsKey(code))
			{
				char c = encodingToCharMapping.get(code);
				decoded += c;
				code = new String(); // wipe the string
			}
		}
		return encodedMsg; // complete this method
	}
}
